### DISCUSSION TITLE
Usage of ezyang/htmlpurifier package

### DISCUSSION THREAD

**Discussion by dmytromikhieiev1985** [PHPOffice/PhpSpreadsheet#3216] [General]
Hi, You have ezyang/htmlpurifier dependency, but I couldn't find how do you exactly use it. This library has LGPL licence which is not very good for our project, so I am wondering is it possible to replace it with something similar

**Comment by MarkBaker** [comment]
It's used to sanitize string values being written to the HTML Writer, so that any html strings in a loaded file can't be used to inject malicious content into html output

We have had a [PR #2498](https://github.com/PHPOffice/PhpSpreadsheet/pull/2498) to replace it with the MIT licensed tgalopin/html-sanitizer; but a PR that failed a significant number of tests.

As a relatively low priority change, I've made no effort to resolve those issues; but feel free to take a look at that PR and see if you can resolve the problems; or to submit a better alternative.

**Reply by dmytromikhieiev1985** [reply]
Thanks for reply. I'll take a look

**Reply by KminekMatej** [reply]
Hi, did you managed to take a look, or perhaps solve it? We are facing the same issue with our product

**Reply by dmytromikhieiev1985** [reply]
Hi, here is PR https://github.com/PHPOffice/PhpSpreadsheet/pull/3695

**Comment by oleibman** [comment]
Closing. Resolved bu PR #3724.
